pyvips/vimage.py

```diff
@@ -899,6 +1000,230 @@ def toarray(self):
         return [[self(x, y) for x in range(self.width)] 
```

Comment:
This will be excruciatingly slow for large images, since it'll end up running a full pipeline for each pixel. Instead, we should render the image with `write_to_memory`, then loop over that creating a python list of lists.

Or perhaps rely on your new array mechanism instead. `toarray()` hasn't been in a released version of pyvips, so it's not too late to simply throw it away.

Reply:
My vote would be to name it as `.tolist()` just like numpy `arr.tolist()` since I think numpy is so ubiquitous these days that `.toarray()` is likely to be surprising when it returns a list.   Anyway, I'll make it efficient so that it doesn't make a  new pipeline for each pixel.

Reply:
`.tolist` is implemented efficiently now, and it does the right thing for complex-valued arrays too,